shm/Map.cpp

```diff
@@ -1,29 +1,28 @@
 #include "Map.hpp"
 #include <algorithm>
 #include <random>
-#include <iostream>
+#include <cstddef>
 
 Map::Map(){
-    islandsLocations_.reserve(amountIslands);
-    generateIslands();
+    islandsLocations_.reserve(amountOfIslands);
+    void generateIslands(int numIslandsToGenerate);
```

Comment:
that void is unnecessary here. When calling a function you don't need to specify it's return value. Type of passed value is also not needed
```suggestion
    generateIslands(numIslandsToGenerate);
```